### ISSUE TITLE
TTS BCE loss issue since labels is not modified exactly correctly in TTS Transformer

### ISSUE THREAD

**Issue by mechanicalsea** [espnet/espnet#3388]
**Describe the bug**

In [epsnet2/tts/transformer](https://github.com/espnet/espnet/blob/4c7442ad8faad6a6d975337563fe7e1a795c00cb/espnet2/tts/transformer.py#L428) and [espnet/nets/pytorch_backend/e2e_tts_transformer.py](https://github.com/espnet/espnet/blob/4c7442ad8faad6a6d975337563fe7e1a795c00cb/espnet/nets/pytorch_backend/e2e_tts_transformer.py#L756)

```python
labels[:, -1] = 1.0  # make sure at least one frame has 1
```

It is not sure the stop frames when `self.reductor_factor > 1` and should be modified as

```python
import torch

labels = torch.scatter(labels, 1, (olens - 1).unsqueeze(1), 1.0)
```

**Example of the bug**

Given the case that 

```python
olens = torch.LongTensor([5, 3])
labels = torch.FloatTensor([
    [0, 0, 0, 0, 1],
    [0, 0, 1, 1, 1],
])
```

when `reduction_factor == 2`, follow the program, we would have

```python
olens = torch.LongTensor([4, 2])
labels = torch.FloatTensor([
    [0, 0, 0, 1], # correct
    [0, 0, 1, 1], # wrong, ``[0, 1, 1, 1]`` is correct
])
```

Here, we can see `olens` and `labels` are **mismatched**. So this may cause error while computing BCE loss.

Wish the report can be helpful.

**Comment by kan-bayashi** [comment]
Thank you for your report.
You are right, it seems the bug.
I will fix the issue.
